### PR TITLE
feat(api): add command publishing support to H/S commands

### PR DESCRIPTION
**Pull request by X-sam**
# Overview

Implements #10539 
# Changelog

 - Adds Heater Shaker Messages, Commands, and Payloads to the Commands Module, as well as their associated types.
 - Updates protocol API for Heater-Shaker with decorators to issue said commands.
  
# Review requests

I'd appreciate another set of eyes to make sure I haven't missed any cases!

# Risk assessment

Very low-risk- just adds human-readable strings.

### CODE REVIEW

**Review by mcous — COMMENTED** [review]
Looks like this branch needs a `make format` run, if you haven't done so already

File: api/src/opentrons/commands/module_commands.py [hunk 1]
```diff
@@ -150,3 +151,39 @@ def thermocycler_wait_for_lid_temp() -> command_types.ThermocyclerWaitForLidTemp
 def thermocycler_close() -> command_types.ThermocyclerCloseCommand:
     text = "Closing Thermocycler lid"
     return {"name": command_types.THERMOCYCLER_CLOSE, "payload": {"text": text}}
+
+def heater_shaker_set_and_wait_for_temperature(temperature: float) -> command_types.HeaterShakerSetAndWaitForTemperatureCommand:
+    # given rounding precision 2, formats float 1.23456 as 1.23
+    formatted_temp = format(temperature,f'.{utils.HS_GCODE_ROUNDING_PRECISION}f')
```
Comment (sanni-t) [inline api/src/opentrons/commands/module_commands.py]: So @pmoegenburg let me know that we actually truncate the value in firmware. So I guess we should use `math.trunc()` to be precise and leave a note that mentions that we're mimicking firmware behavior

**Review by mcous — APPROVED** [review]
LGTM! Let's get this merged into the main H/S in PAPIv2 branch.

Linting CI is failing due to line length errors. I've removed that check from lint via #10695 because it's a useless check given we have `black` to format code and deal with long lines. Should be good to merge into the feature branch, where we can sync up with edge to make the warnings go away

**Review by sanni-t — APPROVED** [review]
👾
Just need to update the celsius param truncation logic but we can handle it in the main h/s in api branch. Merge away!

**Pull request merged**